models/cm_default.h

```diff
@@ -0,0 +1,328 @@
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef CM_DEFAULT_H
+#define CM_DEFAULT_H
+
+// Includes from nestkernel:
+#include "archiving_node.h"
+#include "event.h"
+#include "nest_types.h"
+#include "universal_data_logger.h"
+
+#include "cm_compartmentcurrents.h"
+#include "cm_tree.h"
+
+namespace nest
+{
+
+/* BeginUserDocs: neuron, compartmental model
+
+Short description
++++++++++++++++++
+
+A neuron model with user-defined dendrite structure.
+Currently, AMPA, GABA or AMPA+NMDA receptors.
+
+Description
++++++++++++
+
+`cm_default` is an implementation of a compartmental model. The structure of the
+neuron -- soma, dendrites, axon -- is user-defined at runtime by adding
+compartments through `nest.SetStatus()`. Each compartment can be assigned
+receptors, also through `nest.SetStatus()`.
+
+The default model is passive, but sodium and potassium currents can be added
+by passing non-zero conductances 'g_Na' and 'g_K' with the parameter dictionary
+when adding compartments. Receptors can be AMPA and/or NMDA (excitatory), and
+GABA (inhibitory). Ion channel and receptor currents to the compartments can be
+customized through NESTML
+
+Usage
++++++
+The structure of the dendrite is user defined. Thus after creation of the neuron
+in the standard manner
+
+.. code-block:: Python
+    cm =  nest.Create('cm_default')
+
+compartments can using the `nest.SetStatus()` function
+
+.. code-block:: Python
+    nest.SetStatus(cm, {"compartments": [{"parent_idx": -1,
+                                          "params": {"e_L": -65.}},
+                                          {"parent_idx": 0,
+                                          "params": {"e_L": -60., "g_C": 0.02}})
+
+Each compartment is assigned an index, corresponding to the order in which they
+were added. Subsequently, compartment indices are used to specify parent
+compartments in the tree, or are used to assign receptors to the compartments.
+By convention, the first compartment is the root (soma), which has no parent.
+In this case, `parent_index` is ``-1``.
+
+To add receptors to the model, also use `nest.SetStatus()`
+
+.. code-block:: Python
+    nest.SetStatus(cm, {"receptors": {"comp_idx": 1,
+                                      "receptor_type": "AMPA",
+                                      "params": {"e_AMPA": 0., "tau_AMPA": 3.})
+
+Similar to compartments, each receptor is assigned an index, starting at 0 and
+corresponding to the order in which they are added. This index is used
+subsequently to connect synapses to the receptor
+
+.. code-block:: Python
+    nest.Connect(pre, cm_model, syn_spec={
+        'synapse_model': 'static_synapse', 'weight': 5., 'delay': 0.5,
+        'receptor_type': 2})
+
+Note:
```

Comment:
Please convert to Restructured Text `note` directives.

Reply:
Implemented in commit 488155582